diff --git a/saveinstance.luau b/saveinstance.luau
--- a/saveinstance.luau
+++ b/saveinstance.luau
@@ -109,3 +109,14 @@
 
-local EXECUTOR_NAME = identify_executor and identify_executor() or ""
+local function identifyExecutorSafe()
+	if not identify_executor then
+		return
+	end
+
+	local ok, name, version = pcall(identify_executor)
+	if ok then
+		return name, version
+	end
+end
+
+local EXECUTOR_NAME = identifyExecutorSafe() or ""
 
@@ -5263,8 +5274,8 @@
 	do
-		local execName, execVersion = "UNKNOWN", "0"
-		if identify_executor then
-			execName, execVersion = identify_executor()
-			if not execVersion then
-				execVersion = "0"
-			end
+		local execName, execVersion = identifyExecutorSafe()
+		if not execName then
+			execName = "UNKNOWN"
+		end
+		if not execVersion then
+			execVersion = "0"
 		end
