diff --git a/saveinstance.luau b/saveinstance.luau
--- a/saveinstance.luau
+++ b/saveinstance.luau
@@ -6446,7 +6446,10 @@
 				end)
 			) or "Unknown"
 
-			local executorName = identify_executor and table.concat({ identify_executor() }, " ") or "Unknown"
+			local executorName = table.concat({ identifyExecutorSafe() }, " ")
+			if executorName == "" then
+				executorName = "Unknown"
+			end
 
 			local metaFooter = table.concat({
 				"\n\n\t\tElapsed time: ",
